packages/get-starknet/src/utils/formatter.test.ts

```diff
@@ -61,3 +61,91 @@ describe('formatCalls', () => {
     expect(result).toStrictEqual(expected);
   });
 });
+
+// Helper function to generate the test params
+/* eslint-disable @typescript-eslint/naming-convention */
+const generateParams = ({
+  compiledClassHash = '0xcompiledClassHash',
+  classHash = '0xclassHash',
+  entryPointsConstructor = [{ selector: '0xconstructorSelector', function_idx: 0 }],
+  entryPointsExternal = [{ selector: '0xexternalSelector', function_idx: 1 }],
+  entryPointsL1Handler = [{ selector: '0xhandlerSelector', function_idx: 2 }],
+  abi = '[{"type":"function","name":"transfer"}]',
+} = {}) => ({
+  compiled_class_hash: compiledClassHash,
+  class_hash: classHash,
+  contract_class: {
+    sierra_program: ['0x1', '0x2'],
+    contract_class_version: '1.0.0',
+    entry_points_by_type: {
+      CONSTRUCTOR: entryPointsConstructor,
+      EXTERNAL: entryPointsExternal,
+      L1_HANDLER: entryPointsL1Handler,
+    },
+    abi,
+  },
+});
+
+// Helper function to generate the expected result
+const generateExpected = ({
+  compiledClassHash = '0xcompiledClassHash',
+  classHash = '0xclassHash',
+  entryPointsConstructor = [{ selector: '0xconstructorSelector', function_idx: 0 }],
+  entryPointsExternal = [{ selector: '0xexternalSelector', function_idx: 1 }],
+  entryPointsL1Handler = [{ selector: '0xhandlerSelector', function_idx: 2 }],
+  abi = '[{"type":"function","name":"transfer"}]',
+} = {}) => ({
+  compiledClassHash,
+  classHash,
+  contract: {
+    sierra_program: ['0x1', '0x2'],
+    contract_class_version: '1.0.0',
+    entry_points_by_type: {
+      CONSTRUCTOR: entryPointsConstructor,
+      EXTERNAL: entryPointsExternal,
+      L1_HANDLER: entryPointsL1Handler,
+    },
+    abi,
+  },
+});
+/* eslint-enable @typescript-eslint/naming-convention */
+
+describe('formatDeclareTransaction', () => {
+  it('converts AddDeclareTransactionParameters to DeclareContractPayload format', () => {
+    const params = generateParams();
+    const expected = generateExpected();
+
+    const result = formatDeclareTransaction(params);
+
+    expect(result).toStrictEqual(expected);
+  });
+
+  it('handles missing optional class_hash correctly', () => {
+    const params = generateParams({ classHash: undefined });
+    const expected = generateExpected({ classHash: undefined });
+
+    const result = formatDeclareTransaction(params);
+
+    expect(result).toStrictEqual(expected);
+  });
+
+  it('handles empty entry_points_by_type correctly', () => {
```

Comment:
not quite sure can we have a better wording on that

1. it is not a empty entry_points_by_type, it is entry_points_by_type with all properties is an empty array
2. the title may need more specific, may be u can use it.each to test each property if they are empty and well handled